presto-main/src/main/java/com/facebook/presto/sql/planner/PlanVerifiersFactory.java

```diff
@@ -0,0 +1,135 @@
+
+    public PlanVerifiersFactory()
+    {
+        this.verifiers = ImmutableList.of(new NoSubqueryExpressionLeft(),
+                new NoApplyNodeLeft());
+    }
+
+    @Override
+    public List<PlanVerifier> get()
+    {
+        return verifiers;
+    }
+
+    public void verify(Plan plan)
+    {
+        for (PlanVerifier verifier : verifiers) {
+            verifier.verify(plan.getRoot());
+        }
+    }
+
+    private class NoSubqueryExpressionLeft
+            implements PlanVerifier
+    {
+        @Override
+        public void verify(PlanNode plan)
+        {
+            ImmutableList.Builder<Expression> expressionsBuilder = ImmutableList.builder();
+            plan.accept(new ExpressionExtractor(), expressionsBuilder);
+            for (Expression expression : expressionsBuilder.build()) {
+                new DefaultTraversalVisitor<Void, Void>()
+                {
+                    @Override
+                    protected Void visitSubqueryExpression(SubqueryExpression node, Void context)
+                    {
+                        throw new SemanticException(SemanticErrorCode.NOT_SUPPORTED, node, "Given subquery is not yet supported");
+                    }
+                }.process(expression, null);
+            }
+        }
+    }
+
+    private final class ExpressionExtractor
+            extends SimplePlanVisitor<ImmutableList.Builder<Expression>>
+    {
+        @Override
+        public Void visitFilter(FilterNode node, ImmutableList.Builder<Expression> context)
+        {
+            context.add(node.getPredicate());
+            return super.visitFilter(node, context);
+        }
+
+        @Override
+        public Void visitProject(ProjectNode node, ImmutableList.Builder<Expression> context)
+        {
+            context.addAll(node.getExpressions());
+            return super.visitProject(node, context);
+        }
+
+        @Override
+        public Void visitTableScan(TableScanNode node, ImmutableList.Builder<Expression> context)
+        {
+            context.add(node.getOriginalConstraint());
+            return super.visitTableScan(node, context);
+        }
+
+        @Override
+        public Void visitValues(ValuesNode node, ImmutableList.Builder<Expression> context)
+        {
+            for (List<Expression> row : node.getRows()) {
+                context.addAll(row);
+            }
+            return super.visitValues(node, context);
+        }
+    }
+
+    private class NoApplyNodeLeft
+            implements PlanVerifier
+    {
+        @Override
+        public void verify(PlanNode plan)
+        {
+            plan.accept(new SimplePlanVisitor()
+            {
+                @Override
+                public Object visitApply(ApplyNode node, Object context)
+                {
+                    if (node.getCorrelation().isEmpty()) {
+                        throw new PrestoException(StandardErrorCode.NOT_SUPPORTED, "Not supported subquery");
```

Comment:
Currently, when correlation is not added to apply nodes, saying that query is uncorrelated may lead to confusion.

Apply node loses information about subquery original location in SQL query text.